### PR TITLE
 Adding type hints with mypy. 

### PR DESCRIPTION
**Pull request by taylorcenters**
Adding type hints with mypy. 
migrating zbwardrive, openear, to their own repo

### CODE REVIEW

File: killerbee/GoodFET.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __init__(self,btaddr):
         
     def write(self,msg):
         """Send traffic."""
-        import time;
+        import time; # type: ignore
```
Comment (rmspeers) [inline killerbee/GoodFET.py]: type hints for things like time and most libraries should be present and thus ignore should not be needed

File: killerbee/daintree.py [hunk 2]
```diff
@@ -50,51 +56,58 @@ def pwrite(self, packet, channel=26, rssi=0):
...
         Close the output packet capture.
         @rtype: None
         '''
+        if self._fh is None:
+            raise Exception('File handle does not exist')
+
         del(self._fh)
 
-    def pnext(self):
+    def pnext(self) -> Optional[list[Any]]:
```
Comment (rmspeers) [inline killerbee/daintree.py:94]: You want to import List from typing, and use `List` not `list` I think

File: killerbee/dev_apimote.py [hunk 3]
```diff
@@ -12,22 +12,27 @@
   - Platform recognition (ApiMote versons)
 '''
 
+from typing import Optional
+from typing import Dict
+from typing import Union
+from typing import Any
```
Comment (rmspeers) [inline killerbee/dev_apimote.py]: Type formatters should fix this, but typically the standard is something like `from typing import Optional, Dict, Union, Any`

File: killerbee/dev_apimote.py [hunk 4]
```diff
@@ -101,19 +112,24 @@ def sniffer_on(self, channel=None, page=0):
...
         self.capabilities.require(KBCapabilities.SNIFF)
+        
+        if self.handle is None:
+            raise Exception("Handle does not exist")
 
         self.handle.RF_promiscuity(1)
         self.handle.RF_autocrc(0)
 
-        if channel != None:
+        if channel is None:
```
Comment (rmspeers) [inline killerbee/dev_apimote.py]: or just use `if channel is not None:`

**Review by rmspeers — CHANGES_REQUESTED** [review]

File: killerbee/daintree.py [hunk 5]
```diff
@@ -50,51 +56,58 @@ def pwrite(self, packet, channel=26, rssi=0):
...
         @type savefile: String
         @param savefile: Daintree SNA packet capture filename to read from.
         @rtype: None.  An exception is raised if the capture file is not in Daintree SNA format.
         '''
         DSNA_HEADER1 = b'#Format=4\r\n'
-        self._fh = open(savefile, "rb")
-        header = self._fh.readline()
+        self._fh: Optional[Any] = open(savefile, "rb")
+        header: bytes = self._fh.readline()
+        print(header)
```
Comment (rmspeers) [inline killerbee/daintree.py]: left over

File: killerbee/daintree.py (hunk 2 above)
Comment (taylorcenters) [reply killerbee/daintree.py:94]: looks like > Python 3.8 can use these built-in types. might change to legacy just to support broader versions

File: killerbee/GoodFET.py (hunk 1 above)
Comment (taylorcenters) [reply killerbee/GoodFET.py]: removed

File: killerbee/dev_bumblebee.py [hunk 6]
```diff
@@ -9,22 +9,22 @@
...
-import struct
-import time
-from array import array
-from datetime import datetime
-from .kbutils import KBCapabilities, makeFCS, bytearray_to_bytes
+from __future__ import print_function # type: ignore
+import sys # type: ignore
+import struct # type: ignore
+import time # type: ignore
+from array import array # type: ignore
```
Comment (rmspeers) [inline killerbee/dev_bumblebee.py:16]: lots of unneeded ignores still that will limit the ability of mypy to check

File: killerbee/dev_sewio.py [hunk 7]
```diff
@@ -9,20 +9,20 @@
...
 #
 
-import os
-import time
-import struct
-import time
+import os # type: ignore
+import time # type: ignore
+import struct # type: ignore
+import time # type: ignore
```
Comment (rmspeers) [inline killerbee/dev_sewio.py:15]: ditto above as an example (i was trying not to comment on every single file where the same issue is)

**Review by rmspeers — CHANGES_REQUESTED** [review]

File: killerbee/dev_bumblebee.py (hunk 6 above)
Comment (taylorcenters) [reply killerbee/dev_bumblebee.py:16]: will address these when coverage is added to those files

File: killerbee/daintree.py (hunk 5 above)
Comment (taylorcenters) [reply killerbee/daintree.py]: fixed

**Review by rmspeers — APPROVED** [review]

**Pull request merged**